src/earn/EarnEnterAmount.tsx

```diff
@@ -76,22 +76,44 @@ function useTokens({ pool, mode }: { pool: EarnPosition; mode: EarnDepositMode }
     [swappableTokens, pool.dataProps.depositTokenId, pool.dataProps.withdrawTokenId]
   )
 
-  if (!depositToken) {
+  if (!inputToken) {
     // should never happen
     throw new Error(`Token info not found for token ID ${pool.dataProps.depositTokenId}`)
   }
 
-  return mode === 'deposit' ? [depositToken] : eligibleSwappableTokens
+  if (!withdrawToken) {
+    // should never happen
+    throw new Error(`Token info not found for token ID ${pool.dataProps.withdrawTokenId}`)
+  }
+
+  return {
+    depositTokens: [inputToken],
+    withdrawTokens: [withdrawToken],
+    eligibleSwappableTokens: eligibleSwappableTokens,
+  }
 }
 
 function EarnEnterAmount({ route }: Props) {
   const { t } = useTranslation()
 
   const { pool, mode = 'deposit' } = route.params
-  const tokens = useTokens({ pool, mode })
+  const isWithdrawal = mode === 'withdraw'
+  const { depositTokens, withdrawTokens, eligibleSwappableTokens } = useTokens({ pool })
 
-  const [token, setToken] = useState<TokenBalance>(() => tokens[0])
-  const depositToken = useTokenInfo(pool.dataProps.depositTokenId)
+  const availableInputTokens = useMemo(() => {
+    switch (mode) {
+      case 'deposit':
+      case 'withdraw':
+        return depositTokens
+      case 'swap-deposit':
+        return eligibleSwappableTokens
+    }
+  }, [mode])
+
+  const [inputToken, setInputToken] = useState<TokenBalance>(() => availableInputTokens[0])
+  const [transactionToken, _setTransactionToken] = useState<TokenBalance>(() =>
```

Comment:
Can you check the swap-deposit flow. If this `_setTransactionToken` function is never called then the flow must be broken when the user switches which token they are swapping from.

Reply:
Fixed in 819e297efa5c2cd0d90eac1191be18697442059e!